docs/wasm-assemblyscript.md

```diff
@@ -0,0 +1,20 @@
+# AssemblyScript
+
+[AssemblyScript] 0.10.0 includes support for targeting WASI. To use it, add
+`import "wasi"` at the top of your entrypoint file.
+
+To create a program which can be run directly as a command, pass `--runtime half`
+to the AssemblyScript linker. This selects the [half runtime], which ensures that
+the generated wasm module doesn't contain any extraneous exports. (This isn't
+strictly required today, but the handling of extraneous exports may change in
+the future, so it's encouraged. As a bonus, it also reduces code size.)
+
+To create a program which can be loaded as a library and used from other modules,
+no special options are needed.
+
+[Here is a repository containing an example Hello World program][WASI hello world]
```

Comment:
Might be worth inlining the "hello, world" example, as we do for the Rust docs.

Reply:
Yoooo Whaddup @fitzgen ! :smile: 

Yeah I agree, totally inlining this here would be a good move. 

I would add as:

---

For example, to compile an AssemblyScript module with WASI, you could write the following AssemblyScript code:

**hello-world.ts**
```
import "wasi";
import {Console} from "as-wasi";

Console.log('Hello World!\n');
```

Which uses [as-wasi](https://github.com/jedisct1/as-wasi) as a dependency to make working withe AssemblyScript WASI bindings easier. Then, you can run:

`asc hello-world.ts -b hello-world.wasm`

To get a wasm module that will print "Hello World!" to stdout.

---

Or something along those lines :smile:

Reply:
Thanks! I've now copied the content inline and covered it in the doc.